### ISSUE TITLE
subequations environment results in invalid XHTML

### ISSUE THREAD

**Issue by evoludolab** [brucemiller/LaTeXML#1040]
The minimal LaTeX file `subeqs.tex`:
```
\documentclass{article}
\usepackage{amsmath}
\begin{document}
\begin{subequations}
\begin{align}
1 &= 1\\
2 &= 2
\end{align}
\end{subequations}
\end{document}
```
converted to XHTML using
```
latexml --dest=subeqs.xml subeqs.tex ; latexmlpost --dest=subeqs.xhtml subeqs.xml
```
results in invalid XHTML. The errors are 
```
Line 19, Column 20: document type does not allow element "tbody" here
<tbody id="S0.E1.1"><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
```
and the same for `Line 25, Column 20` when checking the syntax on `https://validator.w3.org/#validate-by-upload`.

Note that this is new in `LaTeXML` version 0.8.3. Version 0.8.2 inserted `tbody` in the proper place and just once whereas 0.8.3 seems to add a (misplaced) `tbody` for each sub-equation.

**Comment by dginev** [comment]
Thanks for the report!

**Comment by brucemiller** [comment]
So, xhtml doesn't allow multiple tbody? That's no good! There needs to be a place for an id for each equation, and tbody seems perfect for that. tr isn't sufficient, since some equations may have multiple rows. Maybe I should look into xhtml5?

**Comment by evoludolab** [comment]
Couldn't the label only be added to the `tr` tags that potentially get an equation number? For equations with multiple rows `latex` requires that labels are suppressed with `\nonumber` or similar. Maybe `lateXML` could take advantage of this?

Some further background on my use case: I generate XHTML followed by some 'manual' post-processing and then stitch things together into an ePub v3. (It would be great if ePub switched to html5 and xhtml could get retired. Currently it seems to be a bit of a hybrid e.g. `data-<name>` attributes are not allowed in xhtml but pass ePub validation.)

Unfortunately the ePub validation fails on the multiple tbody-elements. The error message from `epubcheck-4.0.2` is slightly different from `validator.w3.org` above but with the same gist:
```
ERROR(RSC-005): ./<path to file>/<filename>.xhtml(54,22): Error while parsing file 
'element "tbody" not allowed here; expected the element end-tag or element "tfoot" 
or "tr"'.
```

**Comment by dginev** [comment]
Ah, ePub! I was just wondering what was the realistic XHTML use case in 2018, totally forgetting it's crucial for ePub.

Ok, we need to stay compliant then.

**Comment by brucemiller** [comment]
Ugh, ePub! Yeah, so newer xhtml isn't an option. Yeah, the id could be put on the (first) tr, but in general a given "equation" may take multiple rows, so the id only "contains" part of it. Maybe not a problem if the id is only being used for a fragid in html, but would be if you tried to fetch all the "data" associated with a given id. (Yeah, lot's of scare quotes in that sentence!)

OTOH, not sure what the alternative is, other than descending into span-soup and letting CSS sort it out.
Still, it'd require some annoying conditinalizing of the xslt.

**Comment by brucemiller** [comment]
Aha, actually the problem isn't that multiple tbodys are disallowed (they are!), but that there should only be tbody's *or* td's not a mixture. And in this case, there's a pointless tr preceding the 1st tbody.  So, now to figure out how to eliminate that tr!

**Comment by brucemiller** [comment]
So, ,actually the tr is there to hold an id (which is probably not needed); rather than embark on a difficult cleanup of the tree, the easiest fix is to make sure that tr is also wrapped in a tbody.  Should validate now; Thanks for the report!

**Comment by evoludolab** [comment]
Subequations work like a charm - Thanks!

Unfortunately, just now I noticed a small gap in the fix... Consider the minimal code
```
\documentclass{article}
\usepackage{amsmath}
\begin{document}
\begin{align}
1 &= 1\nonumber\\
2 &= 2
\end{align}
\end{document}
```
which again results in the same invalid XHTML with
```
Line 24, Column 18: document type does not allow element "tbody" here
```
However, the example above validates when omitting `\nonumber`. Note that the same applies for `\notag` instead of `\nonumber`.

**Comment by brucemiller** [comment]
Ooops. I think I've thought it through properly this time (538325f8). This should work in both cases :> Thanks for the report(s)

**Comment by evoludolab** [comment]
Nice!

While the topic is still fresh, I just wanted to add that `\intertext` also causes similar issues with the nesting of `tr` and `tbody`. That is, it inserts a `tr` between `tbody`'s instead of also wrapping it in a `tbody` and consequently does not validate. (I failed to catch that previously because it had been buried in all the other validation errors.)

I tried to outsmart latexml by replacing `\intertext{some text}` by `\noalign{\bigskip\noindent some text\bigskip}` but the resulting XHMTL was essentially identical and equally invalid :-)

Anyways, `\intertext` is really not an essential command and is always easy to avoid.
